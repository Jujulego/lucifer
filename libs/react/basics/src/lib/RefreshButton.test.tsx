import React from 'react';
import { act } from 'react-dom/test-utils';
import { createMount, createShallow } from '@material-ui/core/test-utils';

import RefreshButton from '../lib/RefreshButton';

// Setup
let mount: ReturnType<typeof createMount>;
let shallow: ReturnType<typeof createShallow>;

beforeAll(() => {
  mount = createMount();
  shallow = createShallow();
});

// Tests
it('should render correctly (refreshing)', () => {
  // Render
  const wrapper = shallow(
    <RefreshButton refreshing={true} />
  );

  // Check elements
  expect(wrapper).toMatchSnapshot();
});

it('should render correctly (not refreshing)', () => {
  // Render
  const wrapper = shallow(
    <RefreshButton refreshing={false} />
  );

  // Check elements
  expect(wrapper).toMatchSnapshot();
});

it('should react on click', () => {
  const spy = jest.fn();

  // Render
  const wrapper = mount(
    <RefreshButton refreshing={false} onClick={spy} />
  );

  // Get button
  const button = wrapper.find('button');
  expect(button).toHaveLength(1);

  // Test event
  act(() => {
    button.simulate('click');
  });

  expect(spy).toBeCalled();
});
