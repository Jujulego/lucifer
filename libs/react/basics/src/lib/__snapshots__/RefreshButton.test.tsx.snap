// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly (not refreshing) 1`] = `
<WithStyles(ForwardRef(IconButton))
  classes={
    Object {
      "label": "makeStyles-label-1",
    }
  }
  color="inherit"
  disabled={false}
>
  <ForwardRef(Fade)
    in={false}
    mountOnEnter={true}
    unmountOnExit={true}
  >
    <WithStyles(ForwardRef(CircularProgress))
      size={24}
    />
  </ForwardRef(Fade)>
  <ForwardRef(Fade)
    in={true}
  >
    <Memo([object Object]) />
  </ForwardRef(Fade)>
</WithStyles(ForwardRef(IconButton))>
`;

exports[`should render correctly (refreshing) 1`] = `
<WithStyles(ForwardRef(IconButton))
  classes={
    Object {
      "label": "makeStyles-label-1",
    }
  }
  color="inherit"
  disabled={true}
>
  <ForwardRef(Fade)
    in={true}
    mountOnEnter={true}
    unmountOnExit={true}
  >
    <WithStyles(ForwardRef(CircularProgress))
      size={24}
    />
  </ForwardRef(Fade)>
  <ForwardRef(Fade)
    in={false}
  >
    <Memo([object Object]) />
  </ForwardRef(Fade)>
</WithStyles(ForwardRef(IconButton))>
`;
